### PR TITLE
MapRoulette right click menu

### LINKED ISSUE #1430: [MapRoulette Integration] Add left click menu to Map roulette tasks.
### Description

When fixing Map Roulette tasks it is great to click on the task icon, then go over to the window on the left and scroll to the bottom to find the "I Fixed it" button and then scroll down again to find the submit button. BBbbbuuuuuuuttttt!  Wouldn't it be even better to be able to "left click" on the Map Roulette icon and have a small menu pop up to choose one of the four options. Then another window that would pop up to submit.

### PR DESCRIPTION
**Pull request by tannerwuster**
**Right click** on the MapRoulette icon and have a small menu pop up to choose one of the four options.
Screenshot 2024-12-05 at 16 25 04
closes #1430

### CODE REVIEW

**Review by bhousel — CHANGES_REQUESTED** [review]
Wow this is pretty impressive how quickly you built it!

I think it would be ideally better to use the existing edit menu to show things in different modes - I think somebody had also asked for the Rapid accept/ignore commands to go there too.  

But this is immediately useful and I don't want to hold it up to wait for the Edit Menu to be rewritten, something I can do later when I have more time.

I'll do a review with some feedback, looks mostly pretty good 👍

File: modules/modes/SelectMode.js [hunk 1]
```diff
@@ -157,6 +158,7 @@ export class SelectMode extends AbstractMode {
 
     } else if (datum instanceof QAItem && datum.service === 'maproulette') {
       sidebarContent = uiMapRouletteEditor(context).error(datum);
+      uiMapRouletteMenu(context).error(datum);
```
Comment (Bonkles) [inline modules/modes/SelectMode.js:161]:
```suggestion
      let uiSystem = this.context.systems.ui;
      uiSystem.MapRouletteMenu.error(datum);
```

Comment (Bonkles) [reply modules/modes/SelectMode.js:161]: You've already instantiated a new uiMapRouletteMenu object and attached it to the UiSystem via the `MapRouletteMenu` variable- this call completely re-creates the object, so selectMode and UiSystem aren't able to sync up. 

With this change, you should see the _qaItem variable be defined as you take a spin through the body of the menu rendering code.

**Review by bhousel — APPROVED** [review]
Tested this in staging (at least, the best I could with OSM being read-only), and it seems fine.

**Pull request merged**